### DISCUSSION TITLE
Remove array, delete keys

### DISCUSSION THREAD

**Discussion by darthmat** [react-hook-form/react-hook-form#1879] [Q&A]
Hi!

I have a problem with `remove()` function in nested objects. Somehow it delete object keys that are not set as defaultValue.

My codesandbox:
https://codesandbox.io/s/react-hook-form-usefieldarray-qk7be?file=/src/index.js

behavior:

- Append multiple rows
- Delete one row
- `item.base.title` is removed

What i expected:
When use `remove(index)` it remove whole array object at specific index and don't change any other objects.
`item.base.title` supposed to stay and it supposed to not be an input

is there any workaround for it?

**Comment by bluebill1049** [comment]
`useFieldArray` is made for `inputs` not render text/title.

**Reply by bluebill1049** [reply]
if you want to render for text, perhaps you want to keep inside `useState`.

**Reply by darthmat** [reply]
hmm quite strange because if there is no that nested array its working:
https://codesandbox.io/s/react-hook-form-usefieldarray-s6mqk?file=/src/index.js

**Comment by darthmat** [comment]
here i have another example that `useFieldArray` works with render text/title:
https://codesandbox.io/s/boring-stonebraker-n7b0r?file=/src/index.js

problem is only when there is a nested object like:
`[{
              not_nested: "not_nested",
              base: {
                property: "test",
                id_text: "",
                type: "test",
                title: "test",
                sort: "",
                nested: "nested"
              }
            }]`

`not_nested` key on remove() is still there but `nested` key in base object is cleaned.

**Reply by bluebill1049** [reply]
> `not_nested` key on remove() is still there but `nested` key in base object is cleaned.

can you give more detail on this part?

**Reply by darthmat** [reply]
https://codesandbox.io/s/boring-stonebraker-n7b0r?file=/src/index.js
1. click append button multiple time.
2. this will add this object: `{
              not_nested: "not_nested",
              base: {
                property: "test",
                id_text: "",
                type: "test",
                title: "test",
                sort: "",
                nested: "nested"
              }
            }`
3. And it will add 2 paragraphs. First that used data from: "not_nested" and second from: "base.nested"
4. click remove button this will return this object: `{
              not_nested: "not_nested",
              base: {
                property: "test",
                id_text: "",
                type: "test",
                title: "test",
                sort: "",
              }
            }`

so, if fieldArray is not intended for paragraphs, why "not_nested" paragraph is still there, and "base.nested" is removed?
It will be nice if it will not remove paragraph data at all.

**Reply by bluebill1049** [reply]
you are console log `fields` those will just return the input default value, useFieldArray is uncontrolled as well, so input value live in each inputs. `watch` will return existing form values.
